src/devices/Ssd13xx/Ssd1309.cs

```diff
@@ -0,0 +1,227 @@
+
+        /// <summary>
+        /// Initializes new instance of Ssd13069 device that will communicate using SPI
+        /// in a non-traditional "4-wire SPI" mode that uses DC and RST GPIO pins to switch
+        /// between Data/Command instructions and reset behavior.
+        /// A single-chip CMOS OLED/PLED driver with controller for organic/polymer
+        /// light emitting diode dot-matrix graphic display system.
+        /// </summary>
+        /// <param name="spiDevice">The SPI device used for communication.</param>
+        /// <param name="width">Width of the display. Typically 128 pixels</param>
+        /// <param name="height">Height of the display</param>
+        /// <param name="gpioController">Instance of the boards GpioController</param>
+        /// <param name="csGpioPin">GPIO pin for chip-select. Active state is LOW. Does not guarantee support for multiple SPI bus devices.</param>
+        /// <param name="dcGpioPin">GPIO pin for Data/Command control</param>
+        /// <param name="rstGpioPin">GPIO pin for Reset behavior</param>
+        public Ssd1309(SpiDevice? spiDevice, GpioController gpioController, int csGpioPin, int dcGpioPin, int rstGpioPin, int width, int height)
+        : base(spiDevice, width, height)
+        {
+            _gpioController = gpioController;
+
+            _csGpioPin = _gpioController.OpenPin(csGpioPin, PinMode.Output, PinValue.High);
+            _dcGpioPin = _gpioController.OpenPin(dcGpioPin, PinMode.Output, PinValue.Low);
+            _rstGpioPin = _gpioController.OpenPin(rstGpioPin, PinMode.Output, PinValue.High);
+
+            Reset();
+            Initialize();
+        }
+
+        /// <summary>
+        /// Sends command to the device
+        /// </summary>
+        /// <param name="command">Command being sent</param>
+        public virtual void SendCommand(ISsd1309Command command) => SendCommand((ICommand)command);
+
+        /// <summary>
+        /// Sends command to the device
+        /// </summary>
+        /// <param name="command">Command being sent</param>
+        public override void SendCommand(ISharedCommand command) => SendCommand(command);
+
+        /// <summary>
+        /// Send a command to the display controller.
+        /// </summary>
+        /// <param name="command">The command to send to the display controller.</param>
+        private void SendCommand(ICommand command)
+        {
+            Span<byte> commandBytes = command?.GetBytes();
+
+            if (commandBytes is not { Length: >0 })
+            {
+                throw new ArgumentNullException(nameof(command), "Argument is either null or there were no bytes to send.");
+            }
+
+            // TODO: Is this needed at all if data/command is controlled via GPIO instead of a control byte?
+            Span<byte> writeBuffer = SliceGenericBuffer(commandBytes.Length);
+
+            commandBytes.CopyTo(writeBuffer.Slice(0));
+
+            if (_spiDevice != null)
```

Comment:
_spiDevice from the base should be internal and because it's internal, it should become a property so that it make things easier when it comes to accessing it.

Reply:
In the latest commit, I made the I2cDevice internal as well. I'm not a huge fan of the pattern with both devices in the same class, but I plan to add SPI support for more Ssd13xx displays so I will think of some ways to improve that.